### ISSUE TITLE
Jarvis learn to classify the MNIST dataset(+ load models for classification)

### ISSUE THREAD

**Issue by markioan01** [sukeesh/Jarvis#1025]
I would like to make Jarvis to have the ability to classify the Mnist dataset plus the ability for a user to load on him a model that he has download on his computer and then load an image so he can classify the image according to his learning. Like, you load a pre-trained model that you found on the internet that has the ability to classify the President of the U.S.A. and then you load an image of a President and Jarvis tells you who he thiks he is. Ofcorce if you give him a random image it will classify it like it contains a president, but thats on the user. I would love to take this task! Please give me your response ASAP! Thank you!

**Comment by pnhofmann** [comment]
Hey, sure go for it. Sounds fun.
I more or less don't know anything about AI, so would be even more thrilled about having features like this.

**Comment by markioan01** [comment]
Cool, I will do my best! Is there anyone that I can talk to personally and give me some more info about the project and the structure? Maybe you or someone else

**Comment by pnhofmann** [comment]
Hey, yeah... Your entry-point would be really just to create a new file with this content:
https://github.com/sukeesh/Jarvis#quickstart-create-a-new-feature-plugin

If you have any questions, please just comment here.

**Comment by pnhofmann** [comment]
Sorry - hit wrong button ;).

**Comment by markioan01** [comment]
Hello!
So, I have write the code so Jarvis can read an Mnist type image(png, jpg and more) and the Classify it according to my pre-trainded model. I load the image with the CV2 library and take the pixels, create the tensors needed and pass the in to my model. Then we take a prediction of what number our image represents. The accuracy of my model seems to be abbout 97-98%.
My problem now is that I can not understand how I should pass my code inside jarvis. When I try to run the "main" Jarvis scirpt I get a lot of error about missing libraries.
What I need now is a mini help on passing my code inside.
So if we see my script as a function the only think it needs to run is a PATH for the image and it returns a number from 0 to 9.
I would apreciate your help!

**Comment by pnhofmann** [comment]
Hey, sure! If you want, already upload the code you have and create a PR so I can have a look.

 > When I try to run the "main" Jarvis scirpt I get a lot of error about missing libraries.

Do you modify PYTHONPATH?
After you installed python (``python installer.py``) you should always run Jarvis from the virtualenv. That's ``source env/bin/activate`` to change to the python virtual environment with all the packages installed.

**Comment by markioan01** [comment]
Hey! So, I finaly made jarvis to run on my computer and I finished my code! This is a private repository(I added you to have access). As you said, I uploaded there my code to have a look on it. I have 2 python files. The first one is runnable and the seccond one is my try of making Jarvis to work with my script following the instructions on how to add a new feature, but I couldnt make it work.. Have a look on both codes please, run my runnable script an then check my other script, which is going to run on jarvis and tell me what i done wrong..Or I could do my final commit here and then you could change a bit my script so It can be functional with Jarvis. To be clear, my only problem now is the final structure so the script can be called from jarvis(with the @plugin() etc..
The with my code so you can check: https://github.com/markioan01/test-rep-for-jarvis

**Comment by markioan01** [comment]
I cant find where I should do the upload of my files.. Its my first commitment to a public repository..Could you help? I have finished the code

**Comment by pnhofmann** [comment]
Hey, no worries, this project is for first time contributors ;).

First step is always to fork this repository. That's basically this button:

![edit_2023-01-02_12-01-1672660492](https://user-images.githubusercontent.com/30838368/210228144-7fe79edc-64d9-492c-b0de-cafe449df2fb.jpg)

This will create your own private copy of this projects - which you can edit as you like. Like with git or even with the "Add file" / "Edit" feature directly here on github web interface.

And after you did some changes, a button "Create PR" will appear. PR = Pull Request basically asks me, to include the changes you made in your own copy to this project.

**Comment by markioan01** [comment]
I made the pull request..I have 2 python files: One runnable and another that i tried to make it work with jarvis but it didnt.. I read the instructions on how to add a feature with the "@plugin()" etc but it didnt work when i call it through jarvis. Jarvis tell me that he cant find this instruction.. Check my code

**Comment by pnhofmann** [comment]
Torch is > 1GB which definitely needs extra handling in installer. Did make relevant changes in dev, so therefore decided, to merge into new branch ``master+torch`` for now.

Keeping this issue open as reminder, till merged to master - which will happen when I merge dev -> master (hopefully finally soon).
